assignment1/src/main/java/ro/utcn/sd/he/assignment1/controller/CommandLineController.java

```diff
@@ -0,0 +1,314 @@
+            int answerId = scanner.nextInt();
+            scanner.nextLine(); //dump remaining line
+            Answer answer = answerService.findById(answerId);
+            if(userService.getAuthorOf(answer).getUsername().equals(user.getUsername())){
+                print("New text:");
+                String text = scanner.nextLine();
+                answer.setText(text);           //TODO when it is edited, should i change the creation time? or add last edited time?
+                answerService.saveAnswer(answer);
+            } else {
+                print("You can edit only your own questions!");
+            }
+        } else {
+            print("You need to be logged in to edit an answer!");
+        }
+    }
+
+
+    private void handleRegister(){
+        print("enter the username you want to login with:");
+        String username = scanner.next().trim();
+        scanner.nextLine();
+        print("enter a stronk password:");
+        String password = scanner.next().trim();
+        scanner.nextLine();
+        User user = new User(0,username,password,"user",false,0);
+        user = userService.saveUser(user);
+        print("Registered succesfully:\n" + user.toString());
+    }
+
+    public void handleDeleteAnswer(){
+        print("What answer(id) you want to delete:");
+        int id = scanner.nextInt();
+        scanner.nextLine();
+        answerService.deleteAnswer(id);
+    }
+
+    public void handleVote(){
+        if(user == null){
+            print("you need to be logged in to vote!");
+            return;
+        }
+        print("question or answer?");
+        String type = scanner.next().trim();
+        scanner.nextLine(); //drop remaining line
+        print("insert the id of the post you want to vote:");
+        int id = scanner.nextInt();
+        scanner.nextLine();
+        print("+1 or -1?");
+        int vote = scanner.nextInt();
+        scanner.nextLine();
+        if(type.equals("question")){
+            Question question = questionService.findById(id);
+            if(vote == +1){
+                voteService.voteUp(user, question);
+            }
+            else if(vote == -1){
+                voteService.voteDown(user,question);
+            } else {
+                print("invalid vote option");
+            }
+        } else if ( type.equals("answer")){
+            Answer answer = answerService.findById(id);
+            if(vote == +1){
+                voteService.voteUp(user, answer);
+            }
+            else if(vote == -1){
+                voteService.voteDown(user,answer);
+            } else {
+                print("invalid vote option");
+            }
+        } else {
+            print("You can only vote a question or an answer");
+        }
+    }
+
+    private void handleShow(){
+        print("Select question(id):");
+        int questionId = scanner.nextInt();
+        Question question = questionService.findById(questionId);
```

Comment:
You should include all the logic below inside the service layer. As a rule of thumb: your controller should have only the following logic:
 - Reading input
 - Converting the input to the appropriate Java type (e.g. String to int, String to Date, etc).
 - Deciding what service method to call. Usually, you should call **a single** service method.
 - Calling that method.
 - Printing the results.

Regarding how to handle the vote count: you can either have a transient (not-persisted) field on the posts or you can create a `QuestionWithVote` POJO which extends / contains the question and also the vote count.